From 9d0a5a5b32089a856b52ada9e0f466aec4ea7496 Mon Sep 17 00:00:00 2001
From: Francesco Pham <francesco.pham1@h-partners.com>
Date: Thu, 28 May 2026 12:00:00 +0200
Subject: [PATCH] lzma: add lzma_source_x86 source set for x86_64 device
 targets

Mirrors the existing arm64 source set: includes Asm/x86 headers, adds -march=x86-64, builds common_c_source. lzma_shared selects it when target_cpu == x86_64 so the x86_64 device build links the same way as arm64.

Change-Id: Ifbb91b168e579dec0f797c6bf081d9ccce0ffd5b
Signed-off-by: Francesco Pham <francesco.pham1@h-partners.com>
---
 BUILD.gn | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/BUILD.gn b/BUILD.gn
index 6c471c7..e032da6 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -68,6 +68,17 @@ ohos_source_set("lzma_source_arm64") {
   sources += arm64_asm_source
 }
 
+# on device
+ohos_source_set("lzma_source_x86") {
+  configs = [ ":lzma_config_common" ]
+  public_configs = [ ":lzma_config_common" ]
+
+  include_dirs = [ "Asm/x86" ]
+  cflags = [ "-march=x86-64" ]
+
+  sources = common_c_source
+}
+
 # on host
 ohos_source_set("lzma_source_x86_host") {
   configs = [
@@ -126,7 +137,10 @@ ohos_shared_library("lzma_shared") {
     deps = [ ":lzma_source_arm64" ]
   } else if (target_cpu == "riscv64") {
     deps = [ ":lzma_source_riscv64" ]
+  } else if (target_cpu == "x86_64") {
+    deps = [ ":lzma_source_x86" ]
   }
+  
   innerapi_tags = [
     "chipsetsdk_sp_indirect",
     "platformsdk_indirect",
-- 
2.50.1

